web/modules/custom/dbcdk_community/tests/src/Unit/Profile/ProfileRepositoryTest.php

```diff
@@ -0,0 +1,288 @@
+<?php
+/**
+ * @file
+ * Test case for ProfileRepository.
+ */
+
+namespace Drupal\dbcdk_community\Test\Content;
+
+use DBCDK\CommunityServices\Model\CommunityRole;
+use DBCDK\CommunityServices\Model\Profile as ModelProfile;
+use DBCDK\CommunityServices\Model\Quarantine;
+use Drupal\dbcdk_community\Profile\Profile;
+use Drupal\dbcdk_community\Profile\ProfileRepository;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Profile retrieval.
+ *
+ * @group dbcdk_community
+ */
+class ProfileRepositoryTest extends PHPUnit_Framework_TestCase {
+
+  /**
+   * Test profile retrieval by id.
+   */
+  public function testProfileById() {
+    $profile = (new ModelProfile())->setId(1);
+    $profile_api = $this->getMock('\DBCDK\CommunityServices\Api\ProfileApi');
+    $profile_api->method('profileFindOne')->willReturnMap([
+        [json_encode(['where' => ['id' => 1]]), $profile],
+    ]);
+
+    $repo = new ProfileRepository(
+      $profile_api,
+      $this->getMock('\DBCDK\CommunityServices\Api\QuarantineApi'),
+      $this->getMock('\DBCDK\CommunityServices\Api\CommunityRoleApi')
+    );
+    $this->assertEquals(
+      new Profile($profile),
+      $repo->getProfile(1),
+      'Repository does not return expected profile by id'
+    );
+  }
+
+  /**
+   * Test profile retrival with no matching id.
+   */
+  public function testProfileByNonexistingId() {
```

Comment:
You are right. We are only testing that PHPUnit mocks return NULL. I removed the test case.